yl - Create form component for instructor-based search

wq12li opened the pull request:
In this PR, we add a new frontend form component that allows for instructor-based search.

This component is modeled after the existing CourseOverTimeSearchForm component, and is one part of a larger PR at #38 that is now closed.

# Storybook
[The new component](https://ucsb-cs156-s23.github.io/proj-courses-s23-7pm-3/prs/42/storybook/?path=/docs/components-basiccoursesearch-courseovertimeinstructorsearch--default)

# Screenshots
![image](https://github.com/ucsb-cs156-s23/proj-courses-s23-7pm-3/assets/77405911/b95baaa0-ff0e-4b6f-9a7b-53d841b0e878)


Merging this PR closes #41.

wq12li committed:
yl - create form component with stories and tests

nagpalishaan commented:
LGTM

pconrad commented on frontend/src/main/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.js:
```diff
@@ -0,0 +1,83 @@
+  // Stryker disable OptionalChaining
+  const startQtr = systemInfo?.startQtrYYYYQ || "20211";
+  const endQtr = systemInfo?.endQtrYYYYQ || "20214";
+  // Stryker restore OptionalChaining
```
Thank you!

pconrad commented on frontend/src/main/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.js:
```diff
@@ -0,0 +1,83 @@
+  const localStartQuarter = localStorage.getItem("CourseOverTimeInstructorSearch.StartQuarter");
+  const localEndQuarter = localStorage.getItem("CourseOverTimeInstructorSearch.EndQuarter");
+  const localInstructor = localStorage.getItem("CourseOverTimeInstructorSearch.Instructor");
+
```
We need  this to match with the `// Stryker disable all`
```suggestion
// Stryker restore all
```

pconrad commented on frontend/src/main/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.js:
```diff
@@ -0,0 +1,83 @@
+};
+
+
+  // Stryker disable all : Stryker is testing by changing the padding to 0. But this is simply a visual optimization as it makes it look better
```
This should be removed, and if necessary, a test added for the CSS property.  Please see this web page for advice on how to do it:

https://ucsb-cs156.github.io/topics/stryker/stryker_html_attribute_surviving_mutants.html

pconrad commented on frontend/src/stories/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.stories.js:
```diff
@@ -0,0 +1,35 @@
+//TODO: Backend API?
```
Please avoid comments like this in code that we are merging into main.

pconrad reviewed (changes requested):
A few small things... otherwise looks good.

wq12li committed:
yl - WIP fix comments; stryker is failing

wq12li committed:
yl - bandaid fix by restoring Stryker after local storage is used

wq12li commented:
I've dealt with the stray comment in the stories file, and I've removed the `Stryker disable` on the padding. However, I'm not quite sure how to deal with adding the new `restore` line to the component.

If I add the restore immediately after the `getItem()` calls, mutations survive on the following three lines that call the `useState()` function. However, to kill those mutations, I think I need to mock local storage, which seems to be exactly the reason why someone added a `Stryker disable` into the file to begin with.

For the time being, I've moved the `restore` line to after the `useState()` calls to avoid this, but I'm not sure it's the right way to proceed.

Is there a good way that I can properly test the `useState()` calls, or is just skipping over them okay?

pconrad commented on frontend/src/main/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.js:
```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import { Form, Button, Container, Row, Col } from "react-bootstrap";
+
+//import { allTheLevels } from "fixtures/levelsFixtures";
```
We should not have commented out imports in PRs to main.

Please remove this before making a PR.

And code reviewers, please point this out before we get to the staff review.  This will save a lot of time.

pconrad commented on frontend/src/main/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.js:
```diff
@@ -0,0 +1,82 @@
+
+import { useSystemInfo } from "main/utils/systemInfo";
+import SingleQuarterDropdown from "../Quarters/SingleQuarterDropdown";
+// import SingleSubjectDropdown from "../Subjects/SingleSubjectDropdown";
+//import SingleLevelDropdown from "../Levels/SingleLevelDropdown";
+// import { useBackend  } from "main/utils/useBackend";
```
No commented out imports please

pconrad commented on frontend/src/stories/components/BasicCourseSearch/CourseOverTimeInstructorSearchForm.stories.js:
```diff
@@ -0,0 +1,33 @@
+export const Default = Template.bind({});
+
+Default.args = {
+  submitText: "Create",
```
There is no submitText prop (i.e. parameter) for the `CourseOverTimeInstructorSearchForm` so this should not be here.  

The `Default.args` here are supposed to be the args passed into the component.  Your component takes only one prop, called `fetchJSON` so the line below is correct.

pconrad reviewed (changes requested):
A few small items.

wq12li committed:
yl - remove comments; fix stories issue

pconrad reviewed (approved):
LGTM!
